staging/src/k8s.io/component-base/metrics/timing_histogram.go

```diff
@@ -177,6 +177,9 @@ func (v *TimingHistogramVec) WithLabelValuesChecked(lvs ...string) (GaugeMetric,
 		v.LabelValueAllowLists.ConstrainToAllowedList(v.originalLabels, lvs)
 	}
 	ops, err := v.TimingHistogramVec.GetMetricWithLabelValues(lvs...)
+	if ops == nil {
```

Comment:
shouldn't `err` be set if ops is `nil`?

Reply:
I am not sure I understand your question/point.

In the current code base, exactly one of `ops` and `err` is nil and the other is non-nil.  I new line 181 returns the `err`.  I chose to test `ops == nil` on new line 180 because that most directly related to the problem, the fact that the type narrowing fails when `ops` is nil.

Reply:
I'm not sure I understand the motivation for this change.

As you mentioned above, currently if `ops==nil` (without this change), `err` will be non-nil.
So I expect the caller to handle the error, and never access the `ops` in such case.
From that perspective it doesn't matter much what is returned as a first parameter.

So why do we need this change?

Reply:
[OTOH, if this change is critical for some reason I don't yet see, we should add a test for it.]

Reply:
We need to make this change because when `ops == nil`, `ops.(GaugeMetric)` causes a panic.

Reply:
It makes sense to me to add this check, but I would prefer to verify `err != nil` rather than `ops == nil`, the latter one is quite confusing and I had to look at the code again to understand what we were talking about.

Reply:
+1 on @dgrisonnet suggestion, it reads much cleaner this way.

Reply:
Done in latest revision.